### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by viethung7899**
- Create a simple login page
- Incorporate authentication API
- Persist login by storing the token in the local storage (unsafe for now)

### CODE REVIEW

File: src/auth/index.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const baseApi = axios.create({
+  baseURL: "http://chicken.albernihosting.ca",
+})
+
+
+type JWTPayLoad = {
+  iat: number,
+  drupal: {
+    uid: number,
+    role: string[]
```
Comment (andecoa) [inline src/auth/index.ts:34]: role is a string array?

File: src/index.tsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
-const container = document.getElementById("root")
-const root = createRoot(container as HTMLElement)
+const container = document.getElementById("root");
+const queryClient = new QueryClient();
+const root = createRoot(container as HTMLElement);
 root.render(
   <React.StrictMode>
-    <App />
+    <QueryClientProvider client={queryClient}>
```
Comment (andecoa) [inline src/index.tsx:14]: nit: better to keep this file unmodified and do the `QueryClientProvider` wrapper in `App.tsx`

**Review by andecoa — APPROVED** [review]
looks good. Let's merge since I'll need to use the auth token in navigator

File: src/auth/index.ts (hunk 1 above)
Comment (chrishappy) [reply src/auth/index.ts:34]: Correct!

Comment (chrishappy) [reply src/auth/index.ts:34]: **Farmer**

```
{
  "iat": 1677130705,
  "drupal": {
    "uid": "5",
    "role": [
      "authenticated",
      "ovo_farmer"
    ]
  }
}
```

** Technician**
```
{
  "iat": 1677130768,
  "drupal": {
    "uid": "6",
    "role": [
      "authenticated",
      "ovo_technician"
    ]
  }
}
```

File: src/index.tsx (hunk 2 above)
Comment (viethung7899) [reply src/index.tsx]: Sure

**Pull request merged**